src/nvim/drawline.c

```diff
@@ -395,6 +386,26 @@ static int get_sign_attrs(buf_T *buf, linenr_T lnum, SignTextAttrs *sattrs, int
   return num_signs;
 }
 
+static void get_sign_text(char_u **bufp, win_T *wp, linenr_T lnum, int sign_idx, int *attrp,
+                          int cul_attr, SignTextAttrs sattrs[])
+{
+  SignTextAttrs *sattr = sign_get_attr(sign_idx, sattrs, wp->w_scwidth);
+
+  if (sattr != NULL) {
+    *bufp = (char_u *)sattr->text;
```

Comment:
(Unrelated to this PR)
This refactor also made me realize `sattr->text` is persistent so we don't have to copy it for 'statuscolumn' but can point to it instead:
```diff
-  char sign_text[SIGN_SHOW_MAX][32];  // text in %s (sign column)
+  char *sign_text[SIGN_SHOW_MAX];     // text in %s (sign column)
...
- STRCPY(stcp->sign_text[sign_idx], sattr->text);
+ stcp->sign_text[sign_idx] = sattr->text);  // (or in the get_sign_text_call with *bufp)
```